### DISCUSSION TITLE
内容不全

### DISCUSSION THREAD

**Discussion by waynevan** [Algebra-FUN/WeReadScan#10] [General]
抓取的图片只要部分

**Comment by Algebra-FUN** [comment]
对于这个问题，注意两点：
1. webdriver需要以headless模式启动，只有该模式下才能保证完整的截图 
2. 图书内容需要微是信读书已购图书，如果是未购买的图书，微信读书将不会显示完整内容，自然图片抓取的内容会是不全的（这点是为了保证不侵犯版权）

你可以先尝试已购图书

**Comment by waynevan** [comment]
还是不全
1、是无头模式运行的
chrome_options.add_argument('headless')
2、我是用微信读书的无限卡，添加书架的，这算已购图书吧。

**Comment by Algebra-FUN** [comment]
那可以截个图，具体描述一下是怎样的不全

**Comment by waynevan** [comment]
通过WeReadScan：
每部分内容不完整，比如序章
https://imgtu.com/i/6FIXA1
https://s3.ax1x.com/2021/03/02/6FIXA1.png
手工截取：
https://imgtu.com/i/6Fo9je
https://s3.ax1x.com/2021/03/02/6Fo9je.png

**Comment by waynevan** [comment]
通过WeReadScan下载的PDF
https://anonymousfiles.io/m06y8MKM/

**Comment by Algebra-FUN** [comment]
目前截取不全的问题已修复（WeReadScan==0.8.2)，可在新版本中体验效果
感谢你的issue

**Comment by waynevan** [comment]
更新后比之前好很多  
但是还有问题：
1、好多图片没有
2、部分页面大大小小不一样
https://imgtu.com/i/6kQ439

**Comment by waynevan** [comment]
我找了其他版本，你参考看看
https://www.mediafire.com/file/j4exle3ot3tmfcs/%25E4%25BB%258E0%25E5%2588%25B01%25EF%25BC%259ACTFer%25E6%2588%2590%25E9%2595%25BF%25E4%25B9%258B%25E8%25B7%25AF-Nu1L%25E6%2588%2598%25E9%2598%259F.pdf/file

**Comment by waynevan** [comment]
换了本书，卡到这里不动了，重新运行还是过不去
https://ftp.bmp.ovh/imgs/2021/03/6dc3cb3a3097f0b0.png

**Comment by Algebra-FUN** [comment]
巧了，这是因为章节名里"< TContainerBuilder >"，"<",">"作为文件名是invalid的，这是个偶发事件

**Comment by waynevan** [comment]
哦，那你看看，没图片的问题和，页面大大小小的问题
